File: WorkerNode/src/main/java/com/yahoo/sdvornik/sorter/Sorter.java
package com.yahoo.sdvornik.sorter;

import io.netty.buffer.ByteBuf;

import java.util.Arrays;

/**
 * Abstract class for sorting algorithm implementation.
 */
public abstract class Sorter {

    protected final long[] longArr;

    public Sorter(long[] longArr) {
        this.longArr = longArr;
    }

    public abstract long[] sort();

    protected void swap(int i, int j){
        long temp = longArr[i];
        longArr[i] = longArr[j];
        longArr[j] = temp;
    }
}

File: worker/src/main/kotlin/com/yahoo/sdvornik/sorter/Sorter.kt
package com.yahoo.sdvornik.sorter

/**
 * @author Serg Dvornik <sdvornik@yahoo.com>
 */
abstract class Sorter(protected val longArr: LongArray) {

    abstract fun sort(): LongArray

    protected fun swap(i: Int, j: Int) {
        val temp = longArr[i]
        longArr[i] = longArr[j]
        longArr[j] = temp
    }
}